jobs/build/tarball-sources/Jenkinsfile

```diff
@@ -0,0 +1,157 @@
+    buildlib.initialize()
+    buildlib.cleanWorkdir(env.WORKSPACE)
+
+    try {
+        sshagent(['openshift-bot']) {
+            advisory = params.ADVISORY ? Integer.parseInt(params.ADVISORY.toString()) : 0
+            if (advisory == 0) {
+                error "Need to provide an ADVISORY ID"
+            }
+            def components = params.COMPONENTS.replaceAll(',', ' ').split(' ')
+            def filename = "tarball-source-output.txt"
+            def outdir = "/mnt/nfs/home/jenkins/container-sources/"
+            def rcm_guest = params.RCM_GUEST.toString()
+            currentBuild.description = "tarball: ${components}"
+
+            stage("elliott tarball-source") {
+                component_args = ""
+                components.each {
+                    component_args += "--component ${it}"
+                }
+                buildlib.elliott """
+                    tarball-sources create
+                    --out-dir=${outdir}
+                    --force
+                    ${component_args}
+                    ${advisory} > ${filename}
+                """
+            }
+
+            stage("rsync") {
+                cmd = "rsync -avz --chmod=go+rX ${outdir} ${rcm_guest}"
+                commonlib.shell(
+                    script: cmd,
+                    returnAll: true
+                )
+            }
+
+            stage("file ticket to RCM") {
+                // use awk to print out the sigle line cotains ${advisory}
+                // sth like this:
+                // RHEL-7-OSE-4.2/${advisory}/release/logging-fluentd-container-v4.2.26-202003230335.tar.gz 
+                cmd = """awk '/${advisory}/ { save=\$0 }END{ print save }' ${filename}"""
+                verify = commonlib.shell(
+                        script: cmd,
+                        returnStdout: true
+                )
+
+                cmd = """echo '${verify}' |awk -F '-v' '{print \$2}' |awk -F '-' '{print \$1}' """
+                version = commonlib.shell(
+                        script: cmd,
+                        returnStdout: true
+                )
+
+                currentBuild.description += " ocp ${version}"
+                description = """Hi RCM,
+                    The OpenShift ART team needs to provide sources for `${components}` in https://errata.devel.redhat.com/advisory/${advisory}
+
+                    The following sources are uploaded to ${rcm_guest}
+
+                    ${verify}
+
+                    Attaching source tarballs to be published on ftp.redhat.com as in https://projects.engineering.redhat.com/browse/RCMTEMPL-6549
+                    """
+                withCredentials([usernamePassword(
+                    credentialsId: 'rcm-jira-openshift-art-automation',
+                    usernameVariable: 'JIRA_USERNAME',
+                    passwordVariable: 'JIRA_PASSWORD',
+                )]) {
+                    writeFile file:"jira_login.sh", text:'''#!/usr/bin/expect
```

Comment:
wow, what a pain. opportunity to contribute to the tool so that it doesn't require `expect` just to login automatically...

Reply:
which tool? you mean elliott or sth groovy script to wrap it up?

Reply:
thanks, I got what you mean here, I think it should be an upstream PR, I will try to see if it's easy to do.